validator_client/src/http_api/api_secret.rs

```diff
@@ -1,211 +0,0 @@
-use eth2::lighthouse_vc::{PK_LEN, SECRET_PREFIX as PK_PREFIX};
-use filesystem::create_with_600_perms;
-use libsecp256k1::{Message, PublicKey, SecretKey};
-use rand::thread_rng;
-use ring::digest::{digest, SHA256};
-use std::fs;
-use std::path::{Path, PathBuf};
-use warp::Filter;
-
-/// The name of the file which stores the secret key.
-///
-/// It is purposefully opaque to prevent users confusing it with the "secret" that they need to
-/// share with API consumers (which is actually the public key).
-pub const SK_FILENAME: &str = ".secp-sk";
-
-/// Length of the raw secret key, in bytes.
-pub const SK_LEN: usize = 32;
-
-/// The name of the file which stores the public key.
-///
-/// For users, this public key is a "secret" that can be shared with API consumers to provide them
-/// access to the API. We avoid calling it a "public" key to users, since they should not post this
-/// value in a public forum.
-pub const PK_FILENAME: &str = "api-token.txt";
```

Comment:
For backwards compatibility, lets keep loading the contents of the `api-token.txt` and using that as the token. We can just ignore the old `.secp-sk` file which contains the secret key.

To generate new tokens we can use a random string of bytes, rather than `api-token-0xpubkey`